devito/mpi/distributed.py

```diff
@@ -600,17 +616,57 @@ class CustomTopology(tuple):
     """
 
     def __new__(cls, items, input_comm):
+        # Keep track of nstars and already defined decompositions
         nstars = len([i for i in items if i == '*'])
-        if nstars > 0:
-            if input_comm.size % nstars != 0:
-                raise ValueError("Invalid `topology` for given nprocs")
-            if any(i not in ('*', 1) for i in items):
-                raise ValueError("Custom topology must be only 1 or *")
-
-            v = input_comm.size // nstars
-            processed = [i if i == 1 else v for i in items]
-        else:
+        alloc_procs = np.prod([i for i in items if i != '*'])
+        remprocs = int(input_comm.size // alloc_procs)
```

Comment:
should be `-` so not sure how the rest can work

Reply:
Not sure I understand your comment. Do you mean `-` instead of `//` ?

Reply:
Sorry looked at it too quickly ignore this